morango/models.py

```diff
@@ -289,9 +290,9 @@ def _deserialize_store_model(self):
         When deserializing a store model, we look at the deleted flags to know if we should delete the app model.
         Upon loading the app model in memory we validate the app models fields, if any errors occurs we follow
         foreign key relationships to see if the related model has been deleted to propagate that deletion to the target app model.
-        If there are no errors, we save the app model.
-
-        :return: ``bool`` whether the object was successfully deleted or saved
+        We return:
+        (None, True/False) => if the model was deleted successfully or it did not validate
+        (model, True) => if the model validates successfully
```

Comment:
This framing is a bit confusing to me. Perhaps separately explain each piece of the tuple, or... something?

Reply:
Or, if you feel it simplifies it a bit, perhaps switch to Django-style validation where errors trigger an Exception, and then you just need a single return value.

Reply:
Yeah, I think throwing an exception for validation failure, and then just returning model or None, depending on whether it was deleted or not makes the most sense.